custom_components/econnect_metronet/binary_sensor.py

```diff
@@ -131,21 +131,58 @@ def name(self) -> str:
     @property
     def icon(self) -> str:
         """Return the icon used by this entity."""
-        if self._sensor_type == query.SECTORS:
-            return "hass:shield-home-outline"
-        elif self._sensor_type == query.INPUTS:
-            return "hass:electric-switch"
-        else:
-            return "hass:toggle-switch-off-outline"
+        return "hass:electric-switch"
 
     @property
     def is_on(self) -> bool:
         """Return the binary sensor status (on/off)."""
         # TODO: evaluate to expose a device API to get the status of a sensor
         # instead of duplicating this sensor_type check
-        if self._sensor_type == query.SECTORS:
-            return self._device.sectors_armed.get(self._sensor_id) is not None
-        elif self._sensor_type == query.INPUTS:
-            return self._device.inputs_alerted.get(self._sensor_id) is not None
-        else:
-            return False
+        return self._device.inputs_alerted.get(self._sensor_id) is not None
+
+
+class SectorSensor(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a e-connect sector binary sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        unique_id: str,
+        sector_id: int,
+        config: ConfigEntry,
+        name: str,
+        coordinator: DataUpdateCoordinator,
+        device: AlarmDevice,
+        sensor_type: int,
+    ) -> None:
+        """Construct."""
+        super().__init__(coordinator)
+        self.entity_id = generate_entity_id(config, name)
+        self._name = name
+        self._device = device
+        self._unique_id = unique_id
+        self._sector_id = sector_id
+        self._sensor_type = sensor_type
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique identifier."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of this entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the icon used by this entity."""
+        return "hass:shield-home-outline"
+
+    @property
+    def is_on(self) -> bool:
```

Comment:
For all classes we should also test `is_on`.

I think you should add two tests for this one:
1. Test when the entity is turned on
2. Test when the entity is turned off

To test it, you can do something similar to:
```python
def test_binary_sensor_name(hass, config_entry, alarm_entity):
    # Ensure the sensor has the right name
    coordinator = DataUpdateCoordinator(hass, logging.getLogger(__name__), name="econnect_metronet")
    entity = InputSensor("test_id", 1, config_entry, "1 Tamper Sirena", coordinator, alarm_entity, query.INPUTS)
    assert entity.is_on is False
```

e poi:
```python
def test_binary_sensor_name(hass, config_entry, alarm_entity):
    # Ensure the sensor has the right name
    coordinator = DataUpdateCoordinator(hass, logging.getLogger(__name__), name="econnect_metronet")
    entity = InputSensor("test_id", 1, config_entry, "1 Tamper Sirena", coordinator, alarm_entity, query.INPUTS)
    alarm_entity.inputs_armed.update({entity._sector_id: {}})
    assert entity.is_on is True
```

Off the top of my head this is what I think you should do, maybe you can find other ways to achieve the same results! But basically, by default it should be `off`, and if you add the input/sector ID in the device, it should be turned on.